### PR TITLE
Fix a batch of issues flagged by Coverity

### PR DESCRIPTION
**Pull request by kcgen**

### CODE REVIEW

File: src/dos/dos_keyboard_layout.cpp [hunk 1]
```diff
@@ -331,7 +331,11 @@ Bitu keyboard_layout::read_keyboard_file(const char* keyboard_file_name, Bit32s
 			return KEYB_FILENOTFOUND;
 		}
 		if (tempfile) {
-			fseek(tempfile.get(), start_pos + 2, SEEK_SET);
+			if (fseek(tempfile.get(), start_pos + 2, SEEK_SET) != 0) {
+				LOG(LOG_BIOS, LOG_ERROR)
+				("Keyboard layout file %s not found", keyboard_file_name);
+				return KEYB_INVALIDFILE;
+			}
```
Comment (shermp) [inline src/dos/dos_keyboard_layout.cpp:342]: At this point, it's more likely that some other error occurred, and this error will just mask the true error.

The proper thing to do would be to check for `errno`, and log it.

File: src/cpu/cpu.cpp [hunk 2]
```diff
@@ -2261,87 +2261,72 @@ class CPU final : public Module_base {
 		CPU_Cycles=0;
 		CPU_SkipCycleAutoAdjust=false;
 
-		Prop_multival* p = section->Get_multival("cycles");
+		// Sets the value if the string in within the min and max values
+		auto set_if_in_range = [](const std::string &str,
+		                          int &value,
+		                          const int min_value = 0,
```
Comment (kcgen) [inline src/cpu/cpu.cpp:2267]: Change default to 1, and then compare against >=

File: src/cpu/cpu.cpp [hunk 3]
```diff
@@ -2261,87 +2261,72 @@ class CPU final : public Module_base {
...
+		std::string str;
+		CommandLine cmd(0, p->GetSection()->Get_string("parameters"));
+		if (type == "max") {
+			CPU_CycleMax = 0;
+			CPU_CyclePercUsed = 100;
+			CPU_CycleAutoAdjust = true;
+			CPU_CycleLimit = -1;
+			for (auto cmdnum = 1; cmdnum <= cmd.GetCount(); ++cmdnum) {
+				if (cmd.FindCommand(cmdnum, str)) {
+					if (str.find('%') == str.length() - 1) {
```
Comment (kcgen) [inline src/cpu/cpu.cpp:2289]: This checks if the percent character is the last one in the string; use .back for that.

File: src/cpu/cpu.cpp [hunk 4]
```diff
@@ -158,16 +158,6 @@ Bitu CPU_Pop32(void) {
...
-PhysPt SelBase(Bitu sel) {
-	if (cpu.cr0 & CR0_PROTECTION) {
-		Descriptor desc;
-		cpu.gdt.GetDescriptor(sel,desc);
-		return desc.GetBase();
-	} else {
-		return sel<<4;
-	}
-}
-
```
Comment (shermp) [inline src/cpu/cpu.cpp:170]: Did you mean to completely remove this function? If so, the commit name is misleading.

File: src/cpu/cpu.cpp [hunk 5]
```diff
@@ -2271,84 +2261,71 @@ class CPU final : public Module_base {
...
+		if (type == "max") {
+			CPU_CycleMax = 0;
+			CPU_CyclePercUsed = 100;
+			CPU_CycleAutoAdjust = true;
+			CPU_CycleLimit = -1;
+			for (unsigned int cmdnum = 1; cmdnum <= cmd.GetCount(); ++cmdnum) {
+				if (cmd.FindCommand(cmdnum, str)) {
+					if (str.back() == '%') {
+						str.pop_back();
+						set_if_in_range(str, CPU_CyclePercUsed, 0, 105);
```
Comment (kcgen) [inline src/cpu/cpu.cpp:2290]: Move percent ranges to named `constexpr`s

File: src/dos/dos_keyboard_layout.cpp (hunk 1 above)
Comment (kcgen) [reply src/dos/dos_keyboard_layout.cpp]: Thanks @shermp.
The LOG_MSG now also includes the `errno` string.

File: src/cpu/cpu.cpp (hunk 4 above)
Comment (kcgen) [reply src/cpu/cpu.cpp:170]: Yes, it's was added in 2003 but never used, so has sat as a vestigial remnant. 
I've renamed the commit message to clearer: _"Remove an unused function to eliminate an unchecked rcode warning"_

**Review by shermp — APPROVED** [review]

**Comment by kcgen** [comment]
Thanks for the review, @shermp.

**Pull request merged**